core/src/main/java/org/apache/carbondata/core/datastore/page/UnsafeFixLengthColumnPage.java

```diff
@@ -326,9 +327,17 @@ public void encode(PrimitiveCodec codec) {
   }
 
   @Override
-  public byte[] compress(Compressor compressor) {
-    // TODO: use zero-copy raw compression
-    return super.compress(compressor);
+  public byte[] compress(Compressor compressor) throws MemoryException, IOException {
+    // use raw compression and copy to byte[]
+    int inputSize = pageSize << dataType.getSizeBits();
+    int compressedMaxSize = compressor.maxCompressedLength(inputSize);
+    MemoryBlock compressed = UnsafeMemoryManager.allocateMemoryWithRetry(compressedMaxSize);
```

Comment:
need to use UnsafeMemoryAllocator directly.
Because UnsafeMemoryManager maybe use HeapMemoryAllocator.